src/Models/Services/Utils.ts

```diff
@@ -483,6 +483,19 @@ export function sortAscendingOrDescending<T>(
     };
 }
 
+/**
+ * remove duplicate objects from an array
+ * @param array array of objects to operate on
+ * @param key key of the property to use as the discriminator
+ * @returns a new copy of the array with only unique values
+ */
+export function removeDuplicatesFromArray<T>(array: T[], key: keyof T) {
+    const check = new Set<string>();
+    return array.filter(
```

Comment:
Why not `Array.from(check)`?

Reply:
Short answer: I just copied a stack overflow post.
Long answer: I'm not sure you can have a set of objects. Or at least it doesn't help you with getting a distinct list. So instead, we just keep a list of distinct keys and only include items where the key is not found in the set already.

Reply:
Fair enough, LGTM then